src/core/ColorPicker/ColorPicker.tsx

```diff
@@ -44,17 +44,74 @@ export const DefaultColors = [
 export type ColorPickerProps = {
   /**
    * Color Swatches that will be added within the color palette.
+   * Recommended to use `ColorSwatch` components.
    */
   children?: React.ReactNode;
+  /**
+   * Index of the selected color on the color palette.
+   */
+  activeColor?: number;
```

Comment:
I don't understand... According to the story, we expect the user to handle selection state. Then why was this prop added again?

Reply:
Yeah I needed it to access the current active color, I'm not sure how to do that without this prop. Is there an example of roving tabindex somewhere else that I can refer to?

Reply:
Not sure if we are using roving tabindex in some other component, but it's a pretty standard pattern so you can find tons of examples online. The idea is pretty simple: [set tabindex=-1 for all children except currently focused one (which gets 0)](https://developers.google.com/web/fundamentals/accessibility/focus/using-tabindex#:~:text=roving%20tabindex%20works%20by%20setting%20tabindex%20to%20-1%20for%20all%20children%20except%20the%20currently-active%20one). And then you can find 0 easily, without needing an `activeColor` prop.

Alternatively, you might be able to just use [`Document.activeElement`](https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement) if you force focus on active swatch when color picker is opened.

Reply:
Removed for loop and used findIndex instead

Reply:
Somehow the code became even worse. Now you have two instances of `findIndex`, in addition to 3 branches in your if-else. Why is it so complicated?